docs/metrics/iou.md

```diff
@@ -0,0 +1,176 @@
+</p>
+
+
+The IoU metric measures the **intersection** (the number of pixels **common** between the ground truth and prediction masks, **true positive (TP)**) divided by the **union** (the **total** number of pixels present across **both** masks, **TP + false negative (FN) + false positive (FP)**). And here is the formula for the IoU metric for a segmentation mask:
+
+$$
+\text{IoU} \left( \text{A}, \, \text{B} \right) = \frac {\text{TP}} {\text{TP} + \text{FN} + \text{FP}}
+$$
+
+Let’s look at what TP, FP, and FN look like on a segmentation mask:
+
+![An example of segmentation mask with results](../assets/images/metrics-iou-seg-mask-results.jpg)
+<p style="text-align: center; color: gray;">
+    From left to right: the ground truth segmentation mask, the prediction segmentation mask, and the overlay with TP, FP, and FN labeled
+</p>
+
+From the cat image shown above, when you overlay the ground truth and prediction masks, the pixels that belong to both masks are TP. The pixels that only exist in the ground truth mask are FNs, and the pixels that only exist in the prediction mask are FPs. Let's consider the following pixel counts for each category:
+
+<center>
+
+| # True Positives | # False Positives | # False Negatives |
+| --- | --- | --- |
+| 100 | 25 | 75 |
+
+</center>
+
+Then the IoU becomes:
+
+$$
+\begin{align}
+\text{IoU} &= \frac {100} {(100 + 25 + 75)} \\[1em]
+&= 0.5
+\end{align}
+$$
+
+### Set of Labels
+
+The **set of labels** used in multi-label classification tasks is often a [binarized](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.label_binarize.html) list with a number of label elements, for example, let’s say there are three classes, `Airplane`, `Boat`, and `Car`, and a sample is labeled as `Boat` and `Car`. The binary set of labels would then be $[0, 1, 1]$, where each element represents each class, respectively.
+
+Similar to the segmentation mask, the IoU or Jaccard index metric for the ground truth/prediction labels would be the size of the **intersection** of the two sets (the number of labels **common** between two sets, or TP) divided by the size of the **union** of the two sets (the total number of labels present in both sets, **TP + FN + FP**):
+
+$$
+IoU(A, B) = \frac {TP} {TP + FN + FP}
+$$
+
+The IoU for multi-label classification is defined per class. This technique, also known as one-vs-the-rest (OvR), evaluates each class as a binary classification problem. Per-class IoU values can then be aggregated using different [averaging methods](./averaging-methods.md). The popular choice for this workflow is **macro**, so let’s take a look at examples of different averaged IoU/Jaccard index metrics for multi-class multi-label classification:
+
+**Example: macro IoU of ground truth and prediction sets of labels: `Airplane`, `Boat`, `Car`**
+
+$$
+\text{A} = [[1, 1, 1], \, [1, 0, 0], \, [0, 1, 1]]
```

Comment:
This is not very easy for me to follow — are there three samples that we're comparing here? Having this matrix be square makes it challenging to understand what corresponds to what.